### PR TITLE
[4.0] com contact unused fields

### PR DESCRIPTION
**Pull request by brianteeman**
The form for an individual contact defines fields for all the icon images so that you can override them at the contact level as well as the component level.

However the fields are never displayed so they are never used. I have gone back over 8 years in the history and I cant see them being used at any point in that time period.

This pr removes the fields from the xml. No point in maintaining code that is never used

### CODE REVIEW

**Comment by chmst** [comment]
The icons are used in single contacts. Here for example I have added an image as icon-address

grafik

**Comment by brianteeman** [comment]
how? By using the form in the component options? That hasnt changed.

**Comment by chmst** [comment]
Clobal confiuration - options. sry .. will be back later

**Comment by brianteeman** [comment]
Thought so. I spent a stupid amount of time checking this

**Comment by chmst** [comment]
Don't blame me ;) I spent an insane amout of time to find this: it is the marker-class
https://github.com/joomla/joomla-cms/blob/4.0-dev/components/com_contact/tmpl/contact/default_address.php#L27

**Comment by brianteeman** [comment]
They're still global params aren't they?. Unless you have also found out where you can set the params for an individual contact

**Comment by chmst** [comment]
There is no chance for individual contacts. I think that this was long time before my "active time". 
The default icons were somehow knuffy - 1990 style. And web site admins should have own icons for all their contacts (for all contacts).

**Comment by brianteeman** [comment]
> There is no chance for individual contacts. I think that this was long time before my "active time".

As I said I went back 8 years without finding it

> The default icons were somehow knuffy - 1990 style. And web site admins should have own icons for all their contacts (for all contacts).

Agreed and they still can. This PR doesnt change that. The only thing this pr does is to remove a bunch of fields that are defined in xml and nowhere else

**Comment by chmst** [comment]
Removing these fields would be a B/C break. If websites have adde here own icons. If they are removed, the lyouts can be made much easier.

**Comment by brianteeman** [comment]
> Removing these fields would be a B/C break. 

How? they are NOT used. The ones that are used are in the global component options

**Comment by brianteeman** [comment]
Or am I missing something

**Comment by chmst** [comment]
Params are used here: 
https://github.com/joomla/joomla-cms/blob/4.0-dev/components/com_contact/src/View/Contact/HtmlView.php#L266 

Maybe I am missing something? I never use contact without own override, so am not expert here.

**Comment by brianteeman** [comment]
Those icons are defined in the global options for the contact component aren't they?

Thats not the fields this PR is removing.

Please open joomla in your browser. You can set custom icons in the component options. There is nowhere you can set custom icons for an individual contact.

Now apply this PR. Nothing has changed. Therefore the fields in this xml are not used and can be deleted.

**Comment by Quy** [comment]
See if #31435 confirms this.

**Comment by brianteeman** [comment]
bingo

**Comment by chmst** [comment]
My bad - I mixed up the xml files. You are right.

**Comment by Fedik** [comment]
~the language strings also need to be removed or?~

ah I see, it also in component config, still in use

**Comment by brianteeman** [comment]
Now we know the fields that are being removed can we get some tests please

**Comment by Quy** [comment]
Should the info be updated/removed as this is the only place that these settings can be configured?

Should this be fixed in J3 too?

**Comment by chmst** [comment]
@Quy yes, the message is wrong.

**Comment by brianteeman** [comment]
And it was always bad english. ;)

I could remove it but then there is a consistency issue with the other tabs. How about simply

These settings apply to a single contact

**Comment by chmst** [comment]
I don't use this feature, so cannot say where is the advantage if every contact has own icons. You could change the message and say "these settings are for all contacts and cannot be set individually" (in correct english)?

File: administrator/components/com_contact/config.xml [hunk 1]
```diff
@@ -396,7 +396,7 @@
 	<fieldset
 		name="Icons"
 		label="COM_CONTACT_ICONS_SETTINGS"
-		description="COM_CONTACT_FIELD_CONFIG_INDIVIDUAL_CONTACT_DESC"
+		description="COM_CONTACT_ICONS_SETTINGS_DESC"
```
Comment (Quy) [inline administrator/components/com_contact/config.xml:399]: Why not keep the old language key and update its description?

Comment (brianteeman) [reply administrator/components/com_contact/config.xml:399]: because it is used in other places

**Comment by chmst** [comment]
I have tested this item :white_check_mark: successfully on 59413069748e294c806de5fa32c849591d56d4db<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/34418">issues.joomla.org/tracker/joomla-cms/34418</a>.</sub>

**Comment by ChristineWk** [comment]
I have tested this item :white_check_mark: successfully on 59413069748e294c806de5fa32c849591d56d4db<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/34418">issues.joomla.org/tracker/joomla-cms/34418</a>.</sub>

**Comment by ChristineWk** [comment]
with Patch:

screen shot 2021-06-07 at 11 56 34<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/34418">issues.joomla.org/tracker/joomla-cms/34418</a>.</sub>

**Comment by chmst** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/34418">issues.joomla.org/tracker/joomla-cms/34418</a>.</sub>

**Comment by brianteeman** [comment]
thank you

**Pull request merged**